stack-graphs/src/c.rs

```diff
@@ -614,6 +614,118 @@ pub extern "C" fn sg_stack_graph_add_edges(
     }
 }
 
+//-------------------------------------------------------------------------------------------------
+// Source info
+
+/// Contains information about a range of code in a source code file.
+#[repr(C)]
+#[derive(Clone, Copy, Default)]
+pub struct sg_source_info {
+    /// The location in its containing file of the source code that this node represents.
+    pub span: sg_span,
+    /// The kind of syntax entity this node represents (e.g. `function`, `class`, `method`, etc.).
+    pub syntax_type: sg_string_handle,
+    /// The full content of the line containing this node in its source file.
+    pub containing_line: sg_string_handle,
+}
+
+/// All of the position information that we have about a range of content in a source file
+#[repr(C)]
+#[derive(Clone, Copy, Default)]
+pub struct sg_span {
+    pub start: sg_position,
+    pub end: sg_position,
+}
+
+/// All of the position information that we have about a character in a source file
+#[repr(C)]
+#[derive(Clone, Copy, Default)]
+pub struct sg_position {
+    /// The 0-indexed line number containing the character
+    pub line: usize,
+    /// The offset of the character within its containing line
+    pub column: sg_offset,
+    /// The byte offset (within the file) of the line containing the character
+    pub containing_line: sg_bounds,
+    /// The byte offset (within the file) of the line containing the character, with any leading
+    /// and trailing whitespace removed
+    pub trimmed_line: sg_bounds,
+}
+
+/// The offset of a character within a string (typically a line of source code), using several
+/// different units
+///
+/// All offsets are 0-indexed.
+#[repr(C)]
+#[derive(Clone, Copy, Default)]
+pub struct sg_offset {
+    /// The number of UTF-8-encoded bytes appearing before this character in the string
+    pub utf8_offset: usize,
+    /// The number of UTF-16 code units appearing before this character in the string
+    pub utf16_offset: usize,
+}
+
+/// A half-open range identifying a range of characters in a string.
+#[repr(C)]
+#[derive(Clone, Copy, Default)]
+pub struct sg_bounds {
+    /// The index of the first character in the range.
+    pub start: usize,
+    /// The index of the first character _after_ the range.
+    pub end: usize,
```

Comment:
It is unclear to me whether these are UTF-8, UTF-16, or something else? I expected `sg_offset`s here, I think.

Reply:
This data lets us index into the source code buffer to get the line containing the symbol, so we need UTF-8 byte indexes here.  Updated the comments and the name of the type accordingly.